[wasm][aot] error: conflicting types for 'emscripten_set_main_loop'

Hi, I'm linking statically a library from emscripten that contains the function `emscripten_set_main_loop` which I want to call from C#. It receives a function as first argument. However, I'm getting a _conflicting types_ error at compile time.

## Steps to Reproduce

1. Compile emscripten.h into libemscripten.bc

2. Compile the following C# code into `Program.dll`

```c-sharp
using System;
using System.Runtime.InteropServices;

namespace WasmApp
{
  class MonoPInvokeCallbackAttribute : Attribute
  {
    public MonoPInvokeCallbackAttribute(Type t) { }
  }

  class Program
  {
    private const CallingConvention CallConv = CallingConvention.Winapi;

    [UnmanagedFunctionPointer(CallConv)]
    private delegate void doLoop_t();

    [DllImport("libemscripten")]
    private static extern void emscripten_set_main_loop(doLoop_t func, int fps, int simulate_infinite_loop);

    private static void Main()
    {
      emscripten_set_main_loop(do_loop, 0, 0);
    }

    [MonoPInvokeCallback(typeof(doLoop_t))]
    private static void do_loop()
    {
      Console.WriteLine("loop!");
    }
  }
}
```

3. Package the dll and native library:

`mono $WASM_SDK/packager.exe --emscripten-sdkdir=$EMSDK --mono-sdkdir=$WASM_SDK -appdir=bin --builddir=obj --aot --template=runtime.js --link-mode=SdkOnly --pinvoke-libs=libemscripten --native-lib=libemscripten.bc --asset=index.html Program.dll`

4. Build the obj/ directory with ninja.

<!--
You may drag & drop the attachment (repro code/solution, screenshot, etc.) onto the issue.
-->

### Current Behavior

```log
[116/124] bash -c 'source ./emsdk_env.sh && emcc -Oz -g -s USE_ZLIB=1 -s FORCE_FILESYSTEM=1 -s ASSERTIONS=1 -s DISABLE_EXCEPTION_CATCHING=0 -s WASM=1 -s FULL_ES3=1 -s ALLOW_MEMORY_GROWTH=1 -s BINARYEN=1 -s TOTAL_MEMORY=134217728 -s ALIASING_FUNCTION_POINTERS=0 -s NO_EXIT_RUNTIME=1 -s ERROR_ON_UNDEFINED_SYMBOLS=1 -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ccall', 'cwrap', 'setValue', 'getValue', 'UTF8ToString']" -s "EXPORTED_FUNCTIONS=['___cxa_is_pointer_type', '___cxa_can_catch']" -s "DEFAULT_LIBRARY_FUNCS_TO_INCLUDE=['setThrew', 'memset']" -DENABLE_AOT=1 -DCORE_BINDINGS  -DGEN_PINVOKE  -DDRIVER_GEN=1 -I/home/jcarnero/mono/sdks/out/wasm-runtime-release/include/mono-2.0 -c -o driver.o driver.c'
FAILED: driver.o 
bash -c 'source ./emsdk_env.sh && emcc -Oz -g -s USE_ZLIB=1 -s FORCE_FILESYSTEM=1 -s ASSERTIONS=1 -s DISABLE_EXCEPTION_CATCHING=0 -s WASM=1 -s FULL_ES3=1 -s ALLOW_MEMORY_GROWTH=1 -s BINARYEN=1 -s TOTAL_MEMORY=134217728 -s ALIASING_FUNCTION_POINTERS=0 -s NO_EXIT_RUNTIME=1 -s ERROR_ON_UNDEFINED_SYMBOLS=1 -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ccall', 'cwrap', 'setValue', 'getValue', 'UTF8ToString']" -s "EXPORTED_FUNCTIONS=['___cxa_is_pointer_type', '___cxa_can_catch']" -s "DEFAULT_LIBRARY_FUNCS_TO_INCLUDE=['setThrew', 'memset']" -DENABLE_AOT=1 -DCORE_BINDINGS  -DGEN_PINVOKE  -DDRIVER_GEN=1 -I/home/jcarnero/mono/sdks/out/wasm-runtime-release/include/mono-2.0 -c -o driver.o driver.c'
In file included from driver.c:17:
./pinvoke-table.h:7:6: error: conflicting types for 'emscripten_set_main_loop'
void emscripten_set_main_loop (int,int,int);
     ^
/home/jcarnero/mono/sdks/builds/toolchains/emsdk/upstream/emscripten/system/include/emscripten/emscripten.h:69:13: note: previous declaration is here
extern void emscripten_set_main_loop(em_callback_func func, int fps, int simulate_infinite_loop);
            ^
1 error generated.
emcc: error: '/home/jcarnero/mono/sdks/builds/toolchains/emsdk/upstream/bin/clang -target wasm32-unknown-emscripten -D__EMSCRIPTEN_major__=1 -D__EMSCRIPTEN_minor__=39 -D__EMSCRIPTEN_tiny__=16 -D_LIBCPP_ABI_VERSION=2 -Dunix -D__unix -D__unix__ -Werror=implicit-function-declaration -Xclang -nostdsysteminc -Xclang -isystem/home/jcarnero/mono/sdks/builds/toolchains/emsdk/upstream/emscripten/system/include/compat -Xclang -isystem/home/jcarnero/mono/sdks/builds/toolchains/emsdk/upstream/emscripten/system/include -Xclang -isystem/home/jcarnero/mono/sdks/builds/toolchains/emsdk/upstream/emscripten/system/include/libc -Xclang -isystem/home/jcarnero/mono/sdks/builds/toolchains/emsdk/upstream/emscripten/system/lib/libc/musl/arch/emscripten -Xclang -isystem/home/jcarnero/mono/sdks/builds/toolchains/emsdk/upstream/emscripten/system/local/include -Xclang -isystem/home/jcarnero/mono/sdks/builds/toolchains/emsdk/upstream/emscripten/cache/wasm/include -DEMSCRIPTEN -fno-inline-functions -Oz -g -DENABLE_AOT=1 -DCORE_BINDINGS -DGEN_PINVOKE -DDRIVER_GEN=1 -I/home/jcarnero/mono/sdks/out/wasm-runtime-release/include/mono-2.0 -c driver.c -Xclang -isystem/home/jcarnero/mono/sdks/builds/toolchains/emsdk/upstream/emscripten/system/include/SDL -c -o driver.o -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-cxx-exceptions -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr' failed (1)
```

### Expected Behavior

ninja build finishes sucessfully and I can run a server on the bin/ directory

## On which platforms did you notice this

[ ] macOS
[ ] Linux
[ ] Windows
[x] WebAssembly

**Version Used**:

commit d2da9004266821aba70f5dbad44536880454d3a5
Author: Zoltan Varga <vargaz@gmail.com>
Date:   Fri May 22 03:41:19 2020 -0400

--- a/src/mono/mono/mini/mini-generic-sharing.c
+++ b/src/mono/mono/mini/mini-generic-sharing.c
@@ -1253,6 +1253,27 @@ get_wrapper_shared_type_full (MonoType *t, gboolean is_field)
 		return mono_get_int32_type ();
 	case MONO_TYPE_U4:
 		return m_class_get_byval_arg (mono_defaults.uint32_class);
+	case MONO_TYPE_I8:
+#if TARGET_SIZEOF_VOID_P == 8
+		/* Use native int as its already used for byref */
+		return m_class_get_byval_arg (mono_defaults.int_class);
+#else
+		return m_class_get_byval_arg (mono_defaults.int64_class);
+#endif
+	case MONO_TYPE_U8:
+		return m_class_get_byval_arg (mono_defaults.uint64_class);
+	case MONO_TYPE_I:
+#if TARGET_SIZEOF_VOID_P == 8
+		return m_class_get_byval_arg (mono_defaults.int_class);
+#else
+		return m_class_get_byval_arg (mono_defaults.int32_class);
+#endif
+	case MONO_TYPE_U:
+#if TARGET_SIZEOF_VOID_P == 8
+		return m_class_get_byval_arg (mono_defaults.uint64_class);
+#else
+		return m_class_get_byval_arg (mono_defaults.uint32_class);
+#endif
 	case MONO_TYPE_OBJECT:
 	case MONO_TYPE_CLASS:
 	case MONO_TYPE_SZARRAY:
@@ -1309,16 +1330,6 @@ get_wrapper_shared_type_full (MonoType *t, gboolean is_field)
 			t = shared_type;
 		return t;
 	}
-#if TARGET_SIZEOF_VOID_P == 8
-	case MONO_TYPE_I8:
-		return mono_get_int_type ();
-#endif
-#if TARGET_SIZEOF_VOID_P == 4
-	case MONO_TYPE_I:
-		return mono_get_int32_type ();
-	case MONO_TYPE_U:
-		return m_class_get_byval_arg (mono_defaults.uint32_class);
-#endif
 	default:
 		break;
 	}

